File: syJoh/PGS/Lv2/[1차] 캐시.java
import java.util.*;
class Solution {
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        Queue<String> cache = new LinkedList<>();
        if(cacheSize==0){
            answer = cities.length*5;
            return answer;
        }
        for(String city : cities){
            String c = city.toLowerCase();
            if(cache.contains(c)){
                cache.remove(c);
                cache.offer(c);
                answer+=1;
            }else{
                if(cache.size()<cacheSize){
                    cache.offer(c);
                }else{
                    cache.poll();
                    cache.offer(c);
                }
                answer+=5;
            }
        }
        return answer;
    }
}

File: syJoh/PGS/Lv2/[1차] 캐시.py
def solution(cacheSize, cities):
    answer = 0
    cache = []
    for city in cities:
        #캐시크기가 0 이면 모두 miss
        if cacheSize == 0:
            answer += 5*len(cities)
            break
        #소문자로 전환
        c = city.lower()
        if not c in cache: #cache miss
            if len(cache) < cacheSize:
                cache.append(c)
            else:  # 교체
                cache.pop(0)
                cache.append(c)
            answer += 5
        else: #cache hit
            cache.remove(c)
            cache.append(c)
            answer+=1
    return answer
